### ISSUE TITLE
Il2CppInspector.dll missing

### ISSUE THREAD

**Issue by Rodze** [djkaty/Il2CppInspector#34]
I had downloaded the zip and extracted it and used the inspector a few days ago.

Then I want back to using it now and it wasn't opening and I found this error in the Windows event log:

_Description: A .NET Core application failed.
Application: Il2CppInspector.exe
Path: C:\Users\***\Downloads\Il2CppInspector-2.1\Il2CppInspector.exe
Message: The application to execute does not exist: 'C:\Users\****\AppData\Local\Temp\.net\Il2CppInspector\zyyna2tl.cip\Il2CppInspector.dll'._

I had the temp folder with a bunch of files in it, but that specific dll was missing.

I deleted these temp folders, run il2cppinspector.exe and it opened again.

But I'm opening the issue if you wanna investigate what happened. Maybe add a fail safe to generate the dll again if it's missing since it seems to be generated from the exe anyway.

Thanks for your work.

**Comment by djkaty** [comment]
Nothing in the Temp folders is guaranteed to be retained, Windows will automatically delete files that haven't been used for a while. In general it is not recommended to run applications or store files in the temp folders designated by the OS.

The DLL generation is done by .NET Core's runtime; it is not performed by any code I have written as part of the project and is outside the scope of my control.

In a nutshell, don't run apps from folders the OS might purge 🙂

**Comment by Rodze** [comment]
Il2CppInspecot was in the downloads folder as written in the issue report, not in a temp folder. The missing dll file was stored in a temp folder, but since it's automatically created by .NET, I guess there isn't anything the user can do to prevent it from creating the dll in a temp folder.

The idea of the issue was more to see if your app could check for any missing component and throw the respective error instead of just closing without any warning.

Thx again.

**Comment by djkaty** [comment]
Sorry for misunderstanding your original post! But yes, I'm not even sure execution of the user code even starts until .NET has generated or loaded the DLL, so there's not really much I can do about it. Thanks for posting though, if I could fix that I certainly would 😄
